api/proxy/config/enabled_apis/enabled_apis.go

```diff
@@ -0,0 +1,172 @@
+package enabled_apis
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/Layr-Labs/eigenda/api/proxy/common"
+)
+
+// EnabledAPIs is a wrapper type of the APIs set enum and provides
+// encapsulation for the entrypoint processing to make enablement determination
+// through specific boolean accessors.
+type EnabledAPIs struct {
+	apis []API
+}
+
+func (e EnabledAPIs) Count() uint {
+	return uint(len(e.apis))
+}
+
+// ArbCustomDA ... Is Arbitrum Custom DA enabled?
+func (e EnabledAPIs) ArbCustomDA() bool {
+	return e.has(ArbCustomDAServer)
+}
+
+// RestALTDA ... Is the REST ALTDA server enabled?
+func (e EnabledAPIs) RestALTDA() bool {
+	return e.has(Admin) || e.has(OpGenericCommitment) ||
+		e.has(OpKeccakCommitment) || e.has(StandardCommitment)
+}
+
+// RestALTDAOPGeneric ... Is REST server enabled with op-generic commitment mode
+func (e EnabledAPIs) RestALTDAOPGeneric() bool {
+	return e.has(OpGenericCommitment)
+}
+
+// RestALTDAOPKeccak ... Is REST server enabled with op-keccak commitment mode
+func (e EnabledAPIs) RestALTDAOPKeccak() bool {
+	return e.has(OpKeccakCommitment)
+}
+
+// RestALTStandard ... Is REST server enabled with standard commitment mode
+func (e EnabledAPIs) RestALTStandard() bool {
+	return e.has(StandardCommitment)
+}
+
+// RestALTDAWithAdmin ... Is REST server enabled with admin mode
+func (e EnabledAPIs) RestALTDAWithAdmin() bool {
+	return e.has(Admin)
+}
+
+// Metrics ... Are metrics exposed on the metrics server?
+func (e EnabledAPIs) Metrics() bool {
+	return e.has(MetricsServer)
+}
+
+// Check ... Ensures that expression of the enabled API set is correct
+func (e EnabledAPIs) Check() error {
+	if e.Count() == 0 {
+		return fmt.Errorf("expected at least one \"apis.enabled\" value to be provided")
+	}
+
+	if common.ContainsDuplicates(e.apis) {
+		return fmt.Errorf("enabled apis contains duplicate: %+v", e.apis)
+	}
+
+	if e.Metrics() && (!e.RestALTDA() && !e.ArbCustomDA()) {
+		return fmt.Errorf("metrics cannot be enabled unless `arb` and/or `rest` also is")
+	}
```

Comment:
nit: delete this and replace it with a simpler check "either REST or arb custom must be provided. having metrics turned on/off should be orthogonal, and not the main important point.